### PR TITLE
MWPW-142256: Refactor split sections

### PR DESCRIPTION
**Pull request by rl-adobe**
- Remove the splitSection feature from the utils.js.
- Moved the logic to individual components (banner, layouts, promotion, template-list)

Resolves: [MWPW-142256](https://jira.corp.adobe.com/browse/MWPW-142256)

Test URLs:

Banner
- Before: https://www.adobe.com/de/express/feature/video/convert/avi-to-mp4
- After: https://mwpw-142256-refactor-splitsections--express--adobecom.hlx.page/de/express/feature/video/convert/avi-to-mp4?martech=off

Layouts, Promotion
- Before: https://www.adobe.com/express/feature/image/resize
- After: https://mwpw-142256-refactor-splitsections--express--adobecom.hlx.page/express/feature/image/resize?martech=off

Template-list
- Before: https://www.adobe.com/de/express/learn/blog/merry-christmas-whatsapp-images
- After: https://mwpw-142256-refactor-splitsections--express--adobecom.hlx.page/de/express/learn/blog/merry-christmas-whatsapp-images?martech=off

### CODE REVIEW

File: express/scripts/decorate.js [hunk 1]
```diff
@@ -1,8 +1,42 @@
...
+
 export function addTempWrapper($block, blockName) {
   const wrapper = document.createElement('div');
   const parent = $block.parentElement;
   wrapper.classList.add(`${blockName}-wrapper`);
   parent.insertBefore(wrapper, $block);
   wrapper.append($block);
 }
+
+export function unwrapBlock($block, $blockName) {
```
Comment (qiyundai) [inline express/scripts/decorate.js:12]: might as well remove the dollar signs now 😄

**Comment by qiyundai** [comment]
This type of changes really bring joy to my heart

File: express/blocks/layouts/layouts.js [hunk 2]
```diff
@@ -7,6 +8,9 @@ import {
 } from '../shared/masonry.js';
 
 export default function decorate($block) {
+  addTempWrapper($block, 'template-list');
```
Comment (vhargrave) [inline express/blocks/layouts/layouts.js:11]: wrong wrapper! :)

**Review by vhargrave — CHANGES_REQUESTED** [review]

File: express/scripts/decorate.js (hunk 1 above)
Comment (rl-adobe) [reply express/scripts/decorate.js:12]: Should we remove all dollar sign from the variables?

Comment (vhargrave) [reply express/scripts/decorate.js:12]: yes, they can all go

**Comment by qiyundai** [comment]
Something's still not working right. Take a look at these two pages:
https://mwpw-142256-refactor-splitsections--express--adobecom.hlx.page/in/express/feature/image/resize
vs
https://www.adobe.com/in/express/feature/image/resize
Notice that on this branch's version:

- We failed to split the block into its own section from the default content above it thus making the content lose its max-width wrapper
- We are leaving an empty section at the next node

**Comment by rl-adobe** [comment]
> Something's still not working right. Take a look at these two pages: https://mwpw-142256-refactor-splitsections--express--adobecom.hlx.page/in/express/feature/image/resize vs https://www.adobe.com/in/express/feature/image/resize Notice that on this branch's version:
> 
> * We failed to split the block into its own section from the default content above it, thus making the content lose its max-width wrapper
> * We are leaving an empty section at the next node

**Comment by qiyundai** [comment]
@rl-adobe Another thing I just noticed while I was restructuring our project -- we have a decorate.js in the scripts/utils/ already. Do you think we can consolidate your scripts/decorate.js into it?

**Comment by rl-adobe** [comment]
> @rl-adobe Another thing I just noticed while I was restructuring our project -- we have a decorate.js in the scripts/utils/ already. Do you think we can consolidate your scripts/decorate.js into it?

That should be fine I think. Would you move this code to utils/decorate.js or may I take on this PR?

**Comment by qiyundai** [comment]
@rl-adobe Could you please merge stage into you branch again and take the floating-panel related files exactly as-is from stage? It's been rebuilt entirely and isn't involved in these migration compatibility works anymore.

**Comment by qiyundai** [comment]
This is getting very close! Could you please merge stage in one more time and make sure /express/feature/image/remove-background works properly on mobile? There are some errors in the console and it's probably just simple import path fix related to the split-action.js. Thanks!

**Review by vhargrave — APPROVED** [review]

**Review by qiyundai — APPROVED** [review]

**Pull request merged**